pkg/resource/deploy/snapshot.go

```diff
@@ -140,6 +140,14 @@ func (snap *Snapshot) VerifyIntegrity() error {
 			}
 
 			urns[urn] = state
+
+			// Also register this state with all of it's alias URNs.  Note that there is a period of time between
+			// registering a parent resoruce and registering it's children where the child's `parent` URN will refer to
+			// the aliased name, not the new name.  By the end of a successful deployment, these references should all
+			// be updated.
+			for _, alias := range state.Aliases {
+				urns[alias] = state
+			}
```

Comment:
I'll admit, realizing that we needed this makes me nervous.  I have not been able to trigger any problems this loosening of the checkpoint integrity causes - but it does feel worrisome that there is now a period of time where `parent` (and probably `dependsOn` and maybe other) references stored in the checkpoint will not resolve correctly if looked up directly.  Technically, all the information is present to look them up via `aliases`, but that's not what any normal code will do when interpreting a checkpoint file.  @pgavlin Thoughts on what we should do here?

Reply:
> Thoughts on what we should do here?

This brings to mind another potential issue:
1. The checkpoint contains a resource with URN A and a resource with URN B that is a child of the former.
2. A resource with URN C that is aliased to URN A is registered.
3. A resource with URN A is registered.
4. A resource with URN B is registered that is a child of the resource with URN C.

When the checkpoint is written between steps (3) and (4), the new resource with URN A will appear to be the parent of the old resource with URN B.

I believe that similar problems will crop up with dependencies, providers, etc. Those issues would be much more worrisome, as that information is required for correctness. The more I think about it, the more I am convinced that we will either need to fix up URN references in the checkpoint before writing it if any aliases have been used, or we will need to fix consumers of that information to consider aliases when reading the checkpoint. At that point, we would need to bump the checkpoint version number, as the alias information would be required for correctness.

Reply:
Per offline conversation with @pgavlin, we decided the best approach would be to normalize any URN references in the snapshot to the new URNs prior to persisting.  This ensures that if anything is interrupted, the snapshot still maintains the invariant that all URN references directly point to real entries in the snapshot resource map.